test/test_c10d.py

```diff
@@ -2518,6 +2518,76 @@ def test_find_unused_parameters(find_unused_parameters, test_default=False):
         except Exception as ex:
             self.fail("Unexpected exception: %s" % ex)
 
+    @requires_gloo()
+    @skip_if_lt_x_gpu(2)
+    def test_global_local_unused_params_grad(self):
+        """
+        By simulating a multi-task training, this test is to make sure:
+        1) DDP does not touch the grad of globally unused parameters.
+        2) DDP does update the grad of locally unused parameters.
+        """
+        class GlobalLocalUnusedParamModule(nn.Module):
+            class Task(nn.Module):
+                def __init__(self):
+                    super().__init__()
```

Comment:
we haven't deprecated py2 support yet. So use `super(Task, self).__init__()` here?

Reply:
Good suggestion. Will make the change.

Reply:
Done with 1bc27e3.